app/models/concerns/has_volume.rb

```diff
@@ -0,0 +1,26 @@
+module HasVolume
+
+  extend ActiveSupport::Concern
+
+  included do
+    enum unit: {
```

Comment:
You can set a default on an enum column in the migration, which should generally be the 0 value. That means you won't have to set the unit every time so you could remove it from the factory and the batch form.